packages/parser/src/schemas/cognito.ts

```diff
@@ -0,0 +1,522 @@
+ *       "preferredRole": "string"
+ *     },
+ *     "clientMetadata": { "string": "string" }
+ *   },
+ *   "response": {
+ *     "claimsAndScopeOverrideDetails": {
+ *       "idTokenGeneration": {
+ *         "claimsToAddOrOverride": { "string": "string" },
+ *         "claimsToSuppress": [ "string", "string" ]
+ *       },
+ *       "accessTokenGeneration": {
+ *         "claimsToAddOrOverride": { "string": "string" },
+ *         "claimsToSuppress": [ "string", "string" ],
+ *         "scopesToAdd": [ "string", "string" ],
+ *         "scopesToSuppress": [ "string", "string" ]
+ *       },
+ *       "groupOverrideDetails": {
+ *         "groupsToOverride": [ "string", "string" ],
+ *         "iamRolesToOverride": [ "string", "string" ],
+ *         "preferredRole": "string"
+ *       }
+ *     }
+ *   }
+ * }
+ * ```
+ * @see {@link https://docs.aws.amazon.com/cognito/latest/developerguide/user-pool-lambda-pre-token-generation.html}
+ */
+export const PreTokenGenerationTriggerSchemaV2 = z.object({
+  request: z.object({
+    userAttributes: z.record(z.string(), z.string()),
+    scopes: z.array(z.string()),
+    groupConfiguration: z.object({
+      groupsToOverride: z.array(z.string()),
+      iamRolesToOverride: z.array(z.string()),
+      preferredRole: z.string(),
+    }),
+    clientMetadata: z.record(z.string(), z.string()),
+  }),
+  response: z.object({
+    claimsAndScopeOverrideDetails: z.object({
+      idTokenGeneration: z.object({
+        claimsToAddOrOverride: z.record(z.any()),
+        claimsToSuppress: z.array(z.string()),
+      }),
+      accessTokenGeneration: z.object({
+        claimsToAddOrOverride: z.record(z.any()),
+        claimsToSuppress: z.array(z.string()),
+        scopesToAdd: z.array(z.string()),
+        scopesToSuppress: z.array(z.string()),
+      }),
+      groupOverrideDetails: z.object({
+        groupsToOverride: z.array(z.string()),
+        iamRolesToOverride: z.array(z.string()),
+        preferredRole: z.string(),
+      }),
+    }),
+  }),
+});
+
+/**
+ * A zod schema for a Cognito Migrate User trigger event.
+ *
+ * @example
+ * ```json
+ * {
+ *   "userName": "string",
+ *   "request": {
+ *     "password": "string",
+ *     "validationData": { "key": "value" },
+ *     "clientMetadata": { "key": "value" }
+ *   },
+ *   "response": {
+ *     "userAttributes": { "key": "value" },
+ *     "finalUserStatus": "string",
+ *     "messageAction": "string",
+ *     "desiredDeliveryMediums": [ "string" ],
+ *     "forceAliasCreation": true,
+ *     "enableSMSMFA": true
+ *   }
+ * }
+ * ```
+ * @see {@link https://docs.aws.amazon.com/cognito/latest/developerguide/user-pool-lambda-migrate-user.html}
+ */
+export const MigrateUserTriggerSchema = z.object({
+  userName: z.string(),
+  request: z.object({
+    password: z.string(),
+    validationData: z.record(z.string(), z.string()),
+    clientMetadata: z.record(z.string(), z.string()),
+  }),
+  response: z.object({
```

Comment:
All the values in this object should be `null` (see comment above), the fields are filled as part of the Lambda function request so they are not present in the event.